mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.java

```diff
@@ -1,13 +1,255 @@
+                    GeoPoint position = toGeoPoint(item.state());
+                    if (position != null) {
+                        setMarker(mapView, position, item, item.label(), canDragMarker,
+                                this);
+                        positions.add(position);
+                    }
+                }
+
+                if (!positions.isEmpty()) {
+                    double north = -90;
+                    double south = 90;
+                    double west = 180;
+                    double east = -180;
+                    for (GeoPoint position : positions) {
+                        north = Math.max(position.getLatitude(), north);
+                        south = Math.min(position.getLatitude(), south);
+
+                        west = Math.min(position.getLongitude(), west);
+                        east = Math.max(position.getLongitude(), east);
+                    }
+
+                    Log.d(TAG, String.format("North %f, south %f, west %f, east %f",
+                            north, south, west, east));
+                    BoundingBox boundingBox = new BoundingBox(north, east, south, west);
+                    int extraPixel = (int) convertDpToPixel(24f, mapView.getContext());
+                    try {
+                        mapView.zoomToBoundingBox(boundingBox, false, extraPixel);
+                    } catch (Exception e) {
+                        Log.d(TAG, "Error applying markers", e);
+                    }
+                    if (!allowScroll) {
+                        mapView.setScrollableAreaLimitLongitude(west, east, extraPixel);
+                        mapView.setScrollableAreaLimitLatitude(north, south, extraPixel);
+                    }
+                }
+            } else {
+                GeoPoint position = toGeoPoint(mBoundItem.state());
+                if (position != null) {
+                    setMarker(mapView, position, mBoundItem, mLabelView.getText(), canDragMarker,
+                            this);
+                    moveCamera(mapView, zoomLevel, position);
+                    if (!allowScroll) {
+                        mapView.setScrollableAreaLimitLatitude(position.getLatitude(),
+                                position.getLatitude(), 0);
+                        mapView.setScrollableAreaLimitLongitude(position.getLongitude(),
+                                position.getLongitude(), 0);
+                    }
+                }
+            }
+        }
+
+        /**
+         * This method converts dp unit to equivalent pixels, depending on device density.
+         *
+         * @param dp A value in dp (density independent pixels) unit. Which we need to convert into pixels
+         * @param context Context to get resources and device specific display metrics
+         * @return A float value to represent px equivalent to dp depending on device density
+         */
+        private static float convertDpToPixel(float dp, Context context){
```

Comment:
I'll use the copy introduced in https://github.com/openhab/openhab-android/pull/1357/files#diff-6a62feee90062d42673ee06ee0626d43R384.